src/main/java/spoon/support/reflect/cu/CompilationUnitImpl.java

```diff
@@ -301,6 +306,31 @@ public SourcePosition getOrCreatePartialSourcePosition() {
 		return myPartialSourcePosition;
 	}
 
+	@Override
+	public ElementSourceFragment getRootSourceFragment() {
+		if (rootFragment == null) {
+			String originSourceCode = getOriginalSourceCode();
+			rootFragment = new ElementSourceFragment(this, null);
+			for (CtImport imprt : getImports()) {
+				rootFragment.addChild(new ElementSourceFragment(imprt, null /*TODO role for import of CU*/));
+			}
+			for (CtType<?> ctType : declaredTypes) {
```

Comment:
The CU can also be used for package declaration or module declaration: in those cases, you don't create the tree?

Reply:
I forgot that cases. As fast solution: I will check them and `throw SpoonException(...not supported...)`. Later we can check if we can create the tree for them too - I have no idea yet if it is easily possible or  needs more effort.

Reply:
It looks ok for me. Maybe it worth opening an issue to avoid forgetting it.